### PR TITLE
Autosize nodes doc and tests

### PR DESCRIPTION
**Pull request by GlennPlou**

### CODE REVIEW

File: plugins/org.eclipse.sirius.doc/doc/specifier/diagrams/Diagrams.textile [hunk 1]
```diff
@@ -537,7 +537,11 @@ Most styles give you some control on the size of the graphical elements. The pro
...
-* _Size Computation Expression_: An expression used to dynamically compute the size of a graphical element. It is evaluated in the context of the target element, and should return an integer.
+* _Size Computation Expression_: An expression used to dynamically compute the size of a graphical element. It is evaluated in the context of the target element, and should return an integer. By default, the value 3 is used to define a size of 30 pixels for the graphic element. 
+** For Containers, the -1 value can be used to enable the "auto size" mode at creation time, it works for all styles applicable to Containers.
+** For WorkspaceImage style, set -1 as Size Computation Expression to use the size of the image as the size of the graphic element.
+** For Nodes, a positive integer is expected.
+*** There is a special case with Square Style and label position set to _node_, having -1 in both width and height will allow to auto-size the node regarding the size computation expression result and the label.
```
Comment (lfasani) [inline plugins/org.eclipse.sirius.doc/doc/specifier/diagrams/Diagrams.textile:544]: *** There is a special case with Square Style and label position set to _node_. Having -1 in both width and height will allow to auto-size the node according the size of the label. The size computation expression will be the minimum size.

File: plugins/org.eclipse.sirius.doc/doc/i18n_changes.html [hunk 2]
```diff
@@ -74,6 +74,9 @@ <h3 id="MessageKeysAdded">Message Keys Added</h3>
 			<li>
 				<code>org.eclipse.sirius.diagram.ui.provider.Messages.MovePinnedElementsAction_toolTipText</code>
 			</li>
+			<li>
+				<code>org.eclipse.sirius.diagram.ui.provider.Messages.PinElementsEclipseAction_tooltip</code>
+			</li>
```
Comment (lfasani) [inline plugins/org.eclipse.sirius.doc/doc/i18n_changes.html:79]: is it an omission from another ticket or is it a mistake?

**Review by lfasani — CHANGES_REQUESTED** [review]

File: plugins/org.eclipse.sirius.doc/doc/specifier/diagrams/Diagrams.textile (hunk 1 above)
Comment (GlennPlou) [reply plugins/org.eclipse.sirius.doc/doc/specifier/diagrams/Diagrams.textile]: done

File: plugins/org.eclipse.sirius.doc/doc/i18n_changes.html (hunk 2 above)
Comment (GlennPlou) [reply plugins/org.eclipse.sirius.doc/doc/i18n_changes.html]: I don't know, maybe a rebase issue. Removed.

File: plugins/org.eclipse.sirius.diagram.ui/src-diag/org/eclipse/sirius/diagram/ui/edit/internal/part/AbstractDiagramNodeEditPartRefreshVisualsOperation.java [hunk 3]
```diff
@@ -254,7 +254,7 @@ private int computeHeight(final int width) {
             height = node.getHeight().intValue() * LayoutUtils.SCALE;
         }
         DDiagramElementQuery query = new DDiagramElementQuery(node);
-        if (height == 0 && (new DNodeQuery(node).allowsVerticalResize() || query.isCollapsed())) {
+        if (height == 0 && (new DNodeQuery(node).allowsVerticalResize() || query.isCollapsed()) && !new DNodeQuery(node).isAutoSize()) {
```
Comment (lfasani) [inline plugins/org.eclipse.sirius.diagram.ui/src-diag/org/eclipse/sirius/diagram/ui/edit/internal/part/AbstractDiagramNodeEditPartRefreshVisualsOperation.java:257]: ok for the change even if the whole method is badly written
we should have 
```
if (node.getStyle() instanceof WorkspaceImage) { ..}
else {
        final int tmpHeight = ((Integer) getStructuralFeatureValue(NotationPackage.eINSTANCE.getSize_Height())).intValue();
        if (tmpHeight > 0) {
            height = tmpHeight;
    } else { 
    xxx
}
}
```

**Review by lfasani — APPROVED** [review]

**Comment by scosta-obeo** [comment]
Not valid on ODCE 11.9.0.202402211421 because the merger came later. Requires newer version for validation.

**Pull request merged**